libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -336,6 +337,68 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return crypto.createHash('sha1').update(shrinkwrapContent).digest('hex');
   }
 
+  /**
+   * Determine whether `pnpm-lock.yaml` contains insecure sha1 hashes.
+   * @internal
+   */
+  private _disallowInsecureSha1(
+    customTipsConfiguration: CustomTipsConfiguration,
+    terminal: ITerminal
+  ): boolean {
+    const { packages } = this._shrinkwrapJson;
+    if (packages) {
+      for (const { resolution } of Object.values(packages)) {
+        if ((resolution as { integrity: string }).integrity.startsWith('sha1')) {
+          terminal.writeErrorLine(
```

Comment:
It'd be really useful to tell people which package this showed up in, and, ideally, tell them all of the packages that have problematic hashes instead of just the first one.

Reply:
@octogonz pointed out that someone can just search for "sha1" in the lockfile. Fair enough.

Reply:
You can search for "sha1" in the YAML file pretty easily.

Actually the really valuable information to print would be the file path of the lockfile, since Rush Subspaces introduces a world of multiple `pnpm-lock.yaml` getting installed at once.  However unfortunately some refactoring is needed to obtain that information here, which is better done in a separate PR and probably as part of the subspace effort.